### PR TITLE
catalog: break dependency on eval package

### LINKED ISSUE #88400: catalog: break `eval` depenedency
**Is your feature request related to a problem? Please describe.**

The introduction of virtual privileges brought with it a dependency between `sql/catalog` and `sql/sem/eval` which is far from necessary. The reason it exists is to synthesize a privilege descriptor for virtual tables. We could do that in the descs collection itself when resolving virtual tables. In fact, it'd be much cleaner if we did.

The other two uses of the `GetPrivilegeDescriptor` method are the `GlobalPrivileges` and the `ExternalConnectionPrivileges`. Whenever either is used, we have the things we need on hand to resolve that into a `PrivilegeDescriptor` and thus `catalog.PrivilegeObject` eagerly as opposed to delaying the call into `CheckUserPrivilege`.

### PR DESCRIPTION
**Pull request by postamar**
This dependency was introduced because virtual tables need to have privilege descriptors synthesized for them instead of having them provided by the table descriptor.

This commit refactors the privilege descriptor retrieval logic to break this dependency. No functionality should be altered.

Fixes #88400.

Release note: None

### CODE REVIEW

File: pkg/sql/syntheticprivilege/external_connection_privilege.go [hunk 1]
```diff
@@ -35,17 +33,19 @@ func (e *ExternalConnectionPrivilege) GetPath() string {
 	return fmt.Sprintf("/externalconn/%s", e.ConnectionName)
 }
 
-// GetPrivilegeDescriptor implements the PrivilegeObject interface.
-func (e *ExternalConnectionPrivilege) GetPrivilegeDescriptor(
-	ctx context.Context, planner eval.Planner,
-) (*catpb.PrivilegeDescriptor, error) {
-	if planner.IsActive(ctx, clusterversion.V22_2SystemPrivilegesTable) {
-		return planner.SynthesizePrivilegeDescriptor(ctx, e.GetPath(),
```
Comment (ajwerner) [inline pkg/sql/syntheticprivilege/external_connection_privilege.go:43]: I don't get it, what now looks up the privileges?

Comment (postamar) [reply pkg/sql/syntheticprivilege/external_connection_privilege.go:43]: It's still the planner, just without the indirection via the `eval.Planner` interface. See `grant_revoke_system.go`.

**Review by ajwerner — APPROVED** [review]
> Reviewed 17 of 36 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92601)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Review by rafiss — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92601)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @postamar)

___
*[`pkg/sql/catalog/BUILD.bazel` line 75 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/92601#-NIhU7Wd8k10zXxMYFOS:-NIhU7Wd8k10zXxMYFOT:b-bv57sn) ([raw file](https://github.com/cockroachdb/cockroach/blob/69abebbb7a0de978f719432bc5076bfffd4224c5/pkg/sql/catalog/BUILD.bazel#L75)):*
> ```Starlark
> disallowed_imports_test(
>     "catalog",
>     disallow_cdeps = True,
> ```

could add the package here to prevent a regression

**Comment by postamar** [comment]
> could add the package here to prevent a regression

Thanks! Done.

**Comment by postamar** [comment]
Thanks for the reviews!

bors r+

**Pull request merged**